libvast/src/system/importer.cpp

```diff
@@ -44,79 +44,41 @@ importer_state::~importer_state() {
 }
 
 caf::error importer_state::read_state() {
-  VAST_TRACE("");
-  id_generators.clear();
-  auto file = dir / "available_ids";
+  auto file = dir / "next_id";
   if (exists(file)) {
-    VAST_DEBUG(self, "reads persistent state from", to_string(file));
+    VAST_VERBOSE(self, "reads persistent state from", to_string(file));
     std::ifstream available{to_string(file)};
-    std::string line;
-    while (std::getline(available, line)) {
-      id i;
-      id last;
-      std::istringstream in{line};
-      if (in >> i >> last) {
-        VAST_DEBUG(self, "found ID range:", i, "to", last);
-        id_generators.emplace_back(i, last);
-      } else {
-        VAST_ERROR(self, "got an invalidly formatted persistence file:",
-                   to_string(file));
-        return ec::parse_error;
-      }
-    }
+    available >> next_id;
+    if (!available.eof())
+      VAST_ERROR(
+        self, "got an invalidly formatted persistence file:", to_string(file));
+  } else {
+    VAST_VERBOSE(self, "did not find a state file at", to_string(file));
   }
   return caf::none;
 }
 
 caf::error importer_state::write_state() {
-  VAST_TRACE("");
-  if (id_generators.empty() || available_ids() == 0)
-    return caf::none;
   if (!exists(dir)) {
     auto result = mkdir(dir);
     if (!result)
       return std::move(result.error());
   }
-  std::ofstream available{to_string(dir / "available_ids")};
-  auto i = id_generators.begin();
-  available << i->i << ' ' << i->last;
-  for (++i; i != id_generators.end(); ++i) {
-    available << '\n' << i->i << ' ' << i->last;
-  }
-  VAST_DEBUG(self, "saved", available_ids(), "available IDs");
+  std::ofstream available{to_string(dir / "next_id")};
```

Comment:
Can you add a `dir.is_writable()` check above this line?

Reply:
I'm against that. `dir` should always be a child of a `vast.db` folder, and that check should only be done for the parent.

Reply:
The `vast.db` may already exist or be created by a system service, and the user may be using the wrong `umask`. I don't think we can make this assumption.